src/Test/HUnit/Lang.hs

```diff
@@ -49,7 +49,7 @@ location = case reverse callStack of
 --
 -- @
 --    if conditionIsMet
---        then IO ()
+--        then return ()
 --        else assertFailure msg
```

Comment:
@nomeata Hey! Thanks for opening this PR.  If we are already at it, how about making it more idiomatic, e.g. `unless  conditionIsMet $ assertFailure message`?

Reply:
🤷. Some readers might not actually know about `Control.Monad.unless`.

Reply:
I would argue that then it's the more important that they learn about it :smile:

Reply:
On second though, the statement in the docs is no longer true, e.g. `assertEqual` can't be expressed by using `assertFailure`.  I'm going to delete that whole sentence.

Reply:
piggybacked into #28